contract_isp/wizard/activate_contract_service.py

```diff
@@ -59,31 +46,90 @@ def _get_service_id(self):
     def activate(self):
         contract_service_obj = self.env['contract.service']
         contract_service = contract_service_obj.browse(self.service_id.id)
-
         activation_date = datetime.date(
             int(self.activation_date[:4]),
             int(self.activation_date[5:7]),
             int(self.activation_date[8:10]))
-
         contract_service.write({
             'activation_date': self.activation_date,
             'state': 'active'
         })
-
         query = [
             ('account_id', '=', self.account_id.id),
             ('state', '=', 'draft')
         ]
         draft_line_ids = contract_service_obj.search(query)
-
         if not draft_line_ids:
             for line in self.account_id.contract_service_ids:
                 if line.activation_line_generated is False:
                     line.create_analytic_line(mode='manual',
                                               date=activation_date)
-
                     if line.analytic_line_type == 'r':
                         line.create_analytic_line(mode='prorata',
                                                   date=activation_date)
+        return True
+
+    @api.constrains('activation_date')
+    def check_future_date(self):
+        today = datetime.datetime.utcnow().date()
+        for wiz in self:
+            act_date = datetime.datetime.strptime(
+                wiz.activation_date, DEFAULT_SERVER_DATETIME_FORMAT,
+            ).date()
+            if act_date > today:
+                raise Warning(_('You cannot activate a service in future'))
+                return False
+        return True
+
+
+class contract_service_deactivate(models.TransientModel):
+
+    _name = 'contract.service.deactivate'
+
+    deactivation_date = fields.Datetime('Deactivation Date',
+                                        default=fields.datetime.now())
+    account_id = fields.Many2one('account.analytic.account', 'Account',
+                                 default=_get_account_id)
+    service_id = fields.Many2one('contract.service', 'Service',
+                                 default=_get_service_id)
+
+    @api.multi
+    def deactivate(self):
+        context = self._context
```

Comment:
@JayVora-SerpentCS I don't get why you assign context 3 times here. Why can't you use `obj.with_context(..).action_desactivate(...)` ?